src/Stripe.net/Infrastructure/Extensions/ServiceExtensions.cs

```diff
@@ -0,0 +1,52 @@
+namespace Stripe.Infrastructure.Extensions
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Stripe;
+    using Stripe.Infrastructure.FormEncoding;
+
+    internal static class ServiceExtensions
+    {
+        /// <summary>Creates the full URL for a request.</summary>
+        /// <typeparam name="T">Entity type returned by the service.</typeparam>
+        /// <param name="service">The service sending the request.</param>
+        /// <param name="options">The request parameters.</param>
+        /// <param name="baseUrl">The base URL for the request.</param>
+        /// <param name="isListMethod">Whether the request is a list request or not.</param>
+        /// <returns>The full URL for the request.</returns>
+        public static string ApplyAllParameters<T>(this Service<T> service, BaseOptions options, string baseUrl, bool isListMethod = false)
+            where T : IStripeEntity
+        {
+            var expansions = service.Expansions(isListMethod);
+            if (options != null)
+            {
+                expansions.AddRange(options.Expand);
+            }
+
+            return FormEncoder.AppendQueries(
+                baseUrl,
+                options?.ToQueryString(true, false),
```

Comment:
Just because at the callsite here it's not obvious what the parameters `(true, false)` mean, what do you think about explicitly naming them like `ToQueryString(includeExtraParams: true, includeExpandParams: false)`?